algorithmic_efficiency/workloads/imagenet_vit/imagenet_jax/models.py

```diff
@@ -95,68 +84,49 @@ class Encoder(nn.Module):
   dropout_rate: float = 0.0
 
   @nn.compact
-  def __call__(self, x, train=True):
-    out = {}
-
+  def __call__(self, x: spec.Tensor, train: bool = True) -> spec.Tensor:
     # Input Encoder
     for lyr in range(self.depth):
       block = Encoder1DBlock(
           name=f'encoderblock_{lyr}',
           mlp_dim=self.mlp_dim,
           num_heads=self.num_heads,
           dropout_rate=self.dropout_rate)
-      x, out[f'block{lyr:02d}'] = block(x, train)
-    out['pre_ln'] = x  # Alias for last block, but without the number in it.
-
-    return nn.LayerNorm(name='encoder_norm')(x), out
-
-
-class MAPHead(nn.Module):
-  """Multihead Attention Pooling."""
-  mlp_dim: Optional[int] = None  # Defaults to 4x input dim
-  num_heads: int = 12
-
-  @nn.compact
-  def __call__(self, x):
-    # TODO(lbeyer): condition on GAP(x)
-    n, _, d = x.shape
-    probe = self.param('probe',
-                       nn.initializers.xavier_uniform(), (1, 1, d),
-                       x.dtype)
-    probe = jnp.tile(probe, [n, 1, 1])
-
-    x = nn.MultiHeadDotProductAttention(
-        num_heads=self.num_heads,
-        kernel_init=nn.initializers.xavier_uniform())(probe, x)
-
-    # TODO(lbeyer): dropout_rate on head?
-    y = nn.LayerNorm()(x)
-    x = x + MlpBlock(mlp_dim=self.mlp_dim)(y)
-    return x[:, 0]
+      x = block(x, train)
+    return nn.LayerNorm(name='encoder_norm')(x)
 
 
 class ViT(nn.Module):
   """ViT model."""
 
-  num_classes: int
+  num_classes: int = 1000
   patch_size: Sequence[int] = (16, 16)
   width: int = 768
   depth: int = 12
   mlp_dim: Optional[int] = None  # Defaults to 4x input dim
   num_heads: int = 12
-  posemb: str = 'sincos2d'  # Can also be "learn"
```

Comment:
General question: should we remove the code for all options which we don't use, to simplify the code?

Reply:
Yes, we can remove the code for all options for simplicity. I will work on it.

Reply:
Thanks!